modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/PredefinedValuesTable.tsx

```diff
@@ -304,23 +283,57 @@ export default function PredefinedValuesTable({
 			});
 		};
 
-		Liferay.on('deletePredefinedValueField', deletePredefinedValueField);
-		Liferay.on('handleAddFields', handleAddFields);
-
-		return () => {
-			Liferay.detach('deletePredefinedValueField');
-			Liferay.detach('handleAddFields');
-		};
+		return handleAddFields;
 	}, [
 		creationLanguageId,
 		currentObjectDefinitionFields,
 		disableRequiredChecked,
-		objectFieldsMap,
 		predefinedValues,
 		setValues,
 		values.parameters,
+		getSelectedFields,
 	]);
 
+	const deletePredefinedValueField = useMemo(() => {
```

Comment:
Same as before: `useCallback` can avoid the function nesting. Have you tried using it?

Reply:
Never tried using this before, I'll take a look for sure!